pkg/sparrow/targets/gitlab.go

```diff
@@ -0,0 +1,209 @@
+// sparrow
+// (C) 2023, Deutsche Telekom IT GmbH
+//
+// Deutsche Telekom IT GmbH and all other contributors /
+// copyright owners license this file to you under the Apache
+// License, Version 2.0 (the "License"); you may not use this
+// file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package targets
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/caas-team/sparrow/pkg/checks"
+	"github.com/caas-team/sparrow/pkg/sparrow/gitlab"
+
+	"github.com/caas-team/sparrow/internal/logger"
+)
+
+var _ TargetManager = &gitlabTargetManager{}
+
+// gitlabTargetManager implements TargetManager
+type gitlabTargetManager struct {
+	targets []checks.GlobalTarget
+	mu      sync.RWMutex
+	done    chan struct{}
+	gitlab  gitlab.Gitlab
+	// the DNS name used for self-registration
+	name string
+	// the interval for the target reconciliation process
+	checkInterval time.Duration
+	// the amount of time a target can be
+	// unhealthy before it is removed from the global target list
+	unhealthyThreshold time.Duration
+	// how often the instance should register itself as a global target
+	registrationInterval time.Duration
+	// whether the instance has already registered itself as a global target
+	registered bool
+}
+
+// NewGitlabManager creates a new gitlabTargetManager
+func NewGitlabManager(g gitlab.Gitlab, name string, checkInterval, unhealthyThreshold, regInterval time.Duration) *gitlabTargetManager {
```

Comment:
I believe there might be too many parameters to this constructor. 

Consider using a config struct, that way we could significantly simplify this.

Reply:
Why does no one read what I've written in the PR description :D. 

Of course the constructor is shit, we have to pass everything ATM because we're not reading the config from a file and we're missing a config object.

Still, since I pushed my commits for the load-config feature in this branch (oops), now this is also resolved. Thanks for pointing it out, next time time I'll add a comment in code as well.